test/e2e/play_kube_test.go

```diff
@@ -1098,6 +1115,58 @@ var _ = Describe("Podman play kube", func() {
 		Expect(label).To(ContainSubstring("unconfined_u:system_r:spc_t:s0"))
 	})
 
+	It("podman play kube should use default infra_image", func() {
+		err := writeYaml(checkInfraImagePodYaml, kubeYaml)
+		Expect(err).To(BeNil())
+
+		kube := podmanTest.Podman([]string{"play", "kube", kubeYaml})
+		kube.WaitWithDefaultTimeout()
+		Expect(kube).Should(Exit(0))
+
+		podInspect := podmanTest.Podman([]string{"inspect", "check-infra-image", "--format", "{{ .InfraContainerID }}"})
+		podInspect.WaitWithDefaultTimeout()
+		infraContainerID := podInspect.OutputToString()
+
+		conInspect := podmanTest.Podman([]string{"inspect", infraContainerID, "--format", "{{ .ImageName }}"})
+		conInspect.WaitWithDefaultTimeout()
+		infraContainerImage := conInspect.OutputToString()
+		Expect(infraContainerImage).To(Equal(config.DefaultInfraImage))
+	})
+
+	It("podman play kube should use customized infra_image", func() {
+		conffile := filepath.Join(podmanTest.TempDir, "container.conf")
+
+		infraImage := "k8s.gcr.io/pause:3.2"
+		err := ioutil.WriteFile(conffile, []byte(fmt.Sprintf("[engine]\ninfra_image=\"%s\"\n", infraImage)), 0644)
+		Expect(err).To(BeNil())
+
+		os.Setenv("CONTAINERS_CONF", conffile)
+		if IsRemote() {
+			podmanTest.RestartRemoteService()
+		}
+
+		err = writeYaml(checkInfraImagePodYaml, kubeYaml)
+		Expect(err).To(BeNil())
+
+		kube := podmanTest.Podman([]string{"play", "kube", kubeYaml})
+		kube.WaitWithDefaultTimeout()
+		Expect(kube).Should(Exit(0))
+
+		podInspect := podmanTest.Podman([]string{"inspect", "check-infra-image", "--format", "{{ .InfraContainerID }}"})
+		podInspect.WaitWithDefaultTimeout()
+		infraContainerID := podInspect.OutputToString()
+
+		conInspect := podmanTest.Podman([]string{"inspect", infraContainerID, "--format", "{{ .ImageName }}"})
+		conInspect.WaitWithDefaultTimeout()
+		infraContainerImage := conInspect.OutputToString()
+		Expect(infraContainerImage).To(Equal(infraImage))
+
+		os.Unsetenv("CONTAINERS_CONF")
+		if IsRemote() {
+			podmanTest.RestartRemoteService()
+		}
```

Comment:
Please remove this and use `defer os.Unsetenv("CONTAINERS_CONF")` after `os.Setenv`. This line will never be reached if the test fails and therefore it will keep the envar and make all following tests fail.

Reply:
if using defer, when to unset the env and restart remote service?

Reply:
You don't have to restart the remote server for unset, `defer os.Unsetenv("CONTAINERS_CONF")` will automatically unset the env once the test function exits.

Reply:
I don't think so, the env will be unset, but the remote server will still use the customized containers.conf if not restarting it.

Reply:
No. The remote server is only used for one `It(...)` block. Please use defer if your test fails it will cause all others to fail as well.

Reply:
Thank you!  Code updated.